### PR TITLE
[Case 6569]: Silent failure is possible during Model Creation

### PR DESCRIPTION
**Pull request by 1P-Cusack**
[FogBugz Case 6569](https://highfidelity.fogbugz.com/f/cases/6569/Create-Model-allows-setting-dynamic-and-Exact-All-Polygons-resulting-in-silent-fail): When creating a model, it was possible to elect Exact collision with the dynamic
    property even though this combination is prohibited.  This combination would silently
    fail rather than notifying the user of the error.

This implements safeguards against the invalid combination:

- If the user elects the Exact collision type and the dynamic field is unchecked, then dynamic field is disabled.
- If the user elects a different collision type and the dynamic filed is unchecked, then the dynamic field is enabled.
- If the user has checked the dynamic field and subsequently elects the Exact collision type, then the selection is void and the previous collision selection is retained; however, an error dialog is presented informing the user of the error.

Also fixes a layout issue which resulted in the Cancel button getting cutoff by the window's edge.

**Testing Plan:**
* Open the Create Menu
* Select the Model option within the Create Tab
* Part 1:
    * Observe that the Cancel button isn't cutoff at the edge of the screen.
* Part 2:
    * Check the Dynamic checkbox (it should turn blue when checked).
    * From the Automatic Collision Types selection list, select a type **_other than_** "Exact - All polygons".
    * Observe that the selection is allowed.
    * From the Automatic Collision Types selection list, select the type "Exact - All polygons".
    * Observe that the selection remains is unchanged, the "Exact - All polygons" selection is blocked.
    * Observe error message indicating why the selection failed.
    * Uncheck the Dynamic checkbox (the blue color/highlight should go away).
    * From the Automatic Collision Types selection list, select the type "Exact - All polygons".
    * Observe that the selection of "Exact - All polygons" selection is allowed.
    * With "Exact - All polygons" still selected, attempt to check the Dynamic checkbox.
    * Observe that the Dynamic checkbox is disabled, it can't be checked.
    * From the Automatic Collision Types selection list, select a type **_other than_** "Exact - All polygons".
    * Observe that the selection is allowed.
    * Observe that the Dynamic checkbox is enabled, and can be checked if desired.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4769/HighFidelity-Beta-PR11772-f5aafc3f01308f082ef209b3365e27c2f2dcdc22.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4769/HighFidelity-Beta-PR11772-f5aafc3f01308f082ef209b3365e27c2f2dcdc22.exe

**Comment by Nex-Pro** [comment]
Testing

**Comment by Nex-Pro** [comment]
### PASS

Open the Create Menu
Select the Model option within the Create Tab
Part 1:
Observe that the Cancel button isn't cutoff at the edge of the screen. **SUCCESS**
Part 2:
Check the Dynamic checkbox (it should turn blue when checked). **SUCCESS**
From the Automatic Collision Types selection list, select a type other than "Exact - All polygons".
Observe that the selection is allowed. **SUCCESS**
From the Automatic Collision Types selection list, select the type "Exact - All polygons".
Observe that the selection remains is unchanged, the "Exact - All polygons" selection is blocked. **SUCCESS**
Observe error message indicating why the selection failed. **SUCCESS**
Uncheck the Dynamic checkbox (the blue color/highlight should go away). **SUCCESS**
From the Automatic Collision Types selection list, select the type "Exact - All polygons". **SUCCESS**
Observe that the selection of "Exact - All polygons" selection is allowed. **SUCCESS**
With "Exact - All polygons" still selected, attempt to check the Dynamic checkbox. **SUCCESS**
Observe that the Dynamic checkbox is disabled, it can't be checked. **SUCCESS**
From the Automatic Collision Types selection list, select a type other than "Exact - All polygons". **SUCCESS**
Observe that the selection is allowed. **SUCCESS**
Observe that the Dynamic checkbox is enabled, and can be checked if desired. **SUCCESS**

File: interface/resources/qml/hifi/tablet/NewModelDialog.qml [hunk 1]
```diff
@@ -139,26 +151,50 @@ Rectangle {
...
-                            "Sphere"]
+                    model: collisionArray
+
+                    onCurrentIndexChanged: {
+                        if (collisionArray[currentIndex] === staticMeshCollisionText) {
+                            
+                            if (dynamic.checked) {
+                                currentIndex = priorIndex;
+
+                                errorMessageBox("Models with Automatic Collisions set to \"" + staticMeshCollisionText + "\" cannot be dynamic.");
```
Comment (misslivirose) [inline interface/resources/qml/hifi/tablet/NewModelDialog.qml]: Styling: line length should be <128 characters

**Comment by 1P-Cusack** [comment]
@misslivirose:  Thanks; I've addressed the style fix.  It should be building soon.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4788/HighFidelity-Beta-PR11772-0e90c92cf5038c732cc2449fd39a4b840da9eee6.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4788/HighFidelity-Beta-PR11772-0e90c92cf5038c732cc2449fd39a4b840da9eee6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4966/HighFidelity-Beta-PR11772-0e90c92cf5038c732cc2449fd39a4b840da9eee6.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4966/HighFidelity-Beta-PR11772-0e90c92cf5038c732cc2449fd39a4b840da9eee6.dmg

**Pull request merged**